### PR TITLE
[cmake-user] Fix and extend

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Fix running on x64-linux (disable wxwidgets due to known conflicts).
  Test `find_package(CURL)`.
  Actually build the generated projects, to detect unresolved targets.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  test port, not in database.

### CODE REVIEW

**Comment by dg0yt** [comment]
App. 3 min build time on windows.

File: scripts/test_ports/cmake-user/portfile.cmake [hunk 1]
```diff
@@ -62,6 +62,9 @@ if("find-package" IN_LIST FEATURES)
     if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
         list(REMOVE_ITEM packages "Curses")
     endif()
+    if(VCPKG_TARGET_IS_LINUX)
```
Comment (JackBoosY) [inline scripts/test_ports/cmake-user/portfile.cmake:65]: Can you please add a comment here to tell us what conflict wxwidegets have?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: scripts/test_ports/cmake-user/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply scripts/test_ports/cmake-user/portfile.cmake:65]: You mean to duplicate the comment from the wxwidgets portfile?

Comment (dg0yt) [reply scripts/test_ports/cmake-user/portfile.cmake:65]: Done.

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**